### PR TITLE
docs(migration): migration guide

### PR DESCRIPTION
**Pull request by ISO50**
see #155

### CODE REVIEW

File: MIGRATION_GUIDE.md [hunk 1]
```diff
@@ -31,13 +31,67 @@
 
 ### TODO undocumented breaking migration steps
```
Comment (ISO50) [inline MIGRATION_GUIDE.md]: Need to remove this line

Comment (ISO50) [reply MIGRATION_GUIDE.md]: We need to cover this point: [Extract scopes from body in mock (#157) ](https://github.com/zalando-incubator/authmosphere/commit/d3961030cf1a5d498b6d960e26f4bb08d3a440a0)

As its breaking an not obvoius.

File: MIGRATION_GUIDE.md [hunk 2]
```diff
@@ -31,13 +31,67 @@
...
+  error?: string | Error | object,
+  message?: string
+}
+```
+
+### New mocking function `mockEndpointWithErrorResponse`
+
+The library now exports `mockEndpointWithErrorResponse` which allows to mock an OAuth endpoint with an error response to be able to test behaviour in error case more precisley:
+
+```
```
Comment (ISO50) [inline MIGRATION_GUIDE.md]: <code>```ts</code>

File: MIGRATION_GUIDE.md (hunk 1 above)
Comment (bzums) [reply MIGRATION_GUIDE.md]: True, this PR is only addressing the first 3 points in the ["Todo list"](https://github.com/zalando-incubator/authmosphere/pull/168/files#diff-697debdb32eda4ce3cf8356fe442602dL32), I will make this explicit with the next commit.

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,21 +1,160 @@
...
+  REFRESH_TOKEN_GRANT = 'refresh_token',
+  CLIENT_CREDENTIALS_GRANT = 'client_credentials'
+}
+```
+
+### Mock tooling
+
+#### `mockAccessTokenEndpoint` respects scopes property
+
+Before this release, `mockAccessTokenEndpoint` always includes `uid` as value of the `scopes` property in the returned token. Now, `mockAccessTokenEndpoint` includes the scopes which were requested by the HTTP request. So a request like:
```
Comment (ISO50) [inline CHANGELOG.md]: Maybe `A request like:`

**Comment by bzums** [comment]
Fix references from readme to changelog and migration guide

**Comment by ISO50** [comment]
👍

**Comment by bzums** [comment]
👍

File: README.md [hunk 4]
```diff
@@ -19,17 +19,14 @@ Currently the following flows are supported:
...
-## Project renaming
-
-The project was renamed from `lib-oauth-tooling` to `authmosphere`. In the course of this renaming versioning was restarted at `1.0.0`.
-
 ## Changelog and Migration
 
-* See the [changelog](./changelog) for more information.
-* See the [migration guide](./changelog) for more information.
+* See the [changelog](./changelog.md) for more information.
+* See the [migration guide](./changelog.md) for more information.
```
Comment (Retro64) [inline README.md]: Both time same link?

Comment (bzums) [reply README.md]: fixed 👍

**Comment by bzums** [comment]
👍

**Comment by ISO50** [comment]
👍

**Pull request merged**